### PR TITLE
Issue 2562: Assert to truncatedexception while trying to read event after retention time period

### LINKED ISSUE #2562: Fix Retention system test failure
**Problem description**
With the recent changes, that went in   `readNextEvent()` should throw `TruncatedException`  after the 
retention time period. The system test expects a null event.

Modify the code to change the expectation.

**Problem location**
https://github.com/pravega/pravega/blob/master/test/system/src/test/java/io/pravega/test/system/RetentionTest.java#L171

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by Prabha-Veerubhotla**
Signed-off-by: PrabhaVeerubhotla <vvlprabha@gmail.com>

**Change log description**
*  Asserts that readNextEvent throws `SegmentTruncatedException` 

**Purpose of the change**
Fixes #2562 

**What the code does**
The system tests currently expects a null event, when a reader asks for the event,  after the expiry of retention time period. But the changes in the PR #2533 ensures a `truncated exception` to be thrown in such cases. 
This discrepancy in the validation of the test needs to be changed.
 
**How to verify it**
System tests should pass

### CODE REVIEW

**Comment by Prabha-Veerubhotla** [comment]
[retention system test failure](https://trello.com/c/NpIYz9J3/43-retention-system-test-failure)

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java [hunk 1]
```diff
@@ -167,14 +163,7 @@ public void retentionTest() throws Exception {
...
         //expectation is it should have been truncated and we should find stream to be empty
-        try {
-            String readEvent = reader.readNextEvent(6000).getEvent();
-            log.debug("Reading event: {} ", readEvent);
-            assertEquals(null, readEvent);
-        }  catch (ReinitializationRequiredException e) {
-            log.error("Unexpected request to reinitialize {}", e);
-            Assert.fail("Unexpected request to reinitialize.Test failed.");
-        }
+        assertThrows(TruncatedDataException.class, () -> reader.readNextEvent(6000));
```
Comment (shiveshr) [inline test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: i think we should call readNextEvent again and verify that it is able to successfully read the data post truncation too.

Comment (Prabha-Veerubhotla) [reply test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: done.

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java [hunk 2]
```diff
@@ -167,15 +164,14 @@ public void retentionTest() throws Exception {
...
+
+        //verify reader functionality is unaffected post truncation
+        String event = "newEvent";
+        writer.writeEvent(event);
+        log.info("Writing event: {}", event);
+        Assert.assertEquals(event, reader.readNextEvent(6000).getEvent());
 
-       log.debug("The stream is already truncated.Simple retention test passed.");
+        log.debug("The stream is already truncated.Simple retention test passed.");
```
Comment (fpj) [inline test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: Remove additional leading space.

Comment (Prabha-Veerubhotla) [reply test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: I have actually corrected the wrong indentation here.

**Pull request merged**